src/cript/nodes/supporting_nodes/file.py

```diff
@@ -152,25 +152,26 @@ def __init__(self, name: str, source: str, type: str, extension: str = "", data_
 
         super().__init__(name=name, notes=notes, **kwargs)
 
-        # always giving the function the required str regardless if the input `Path` or `str`
-        if _is_local_file(file_source=str(source)):
-            # upload file source if local file
-            source = _upload_file_and_get_object_name(source=source)
-
         # TODO check if vocabulary is valid or not
         # is_vocab_valid("file type", type)
 
         # setting every attribute except for source, which will be handled via setter
         self._json_attrs = replace(
             self._json_attrs,
             type=type,
-            source=source,
+            # always giving the function the required str regardless if the input `Path` or `str`
+            source=str(source),
             extension=extension,
             data_dictionary=data_dictionary,
         )
 
         self.validate()
 
+    def ensure_uploaded(self):
+        if _is_local_file(file_source=self.source):
+            # upload file source if local file
+            self.source = _upload_file_and_get_object_name(source=self.source)
```

Comment:
This was not obvious to me what it was doing and I had to do some digging around to figure it out. I strongly recommend docstrings.

Reply:
I added a doc string.